### ISSUE TITLE
Didn

### ISSUE THREAD

**Issue by pussinmood** [mlzxy/arp#8]

**Comment by mlzxy** [comment]
Have you tried evaluating multiple checkpoints?  The last checkpoint sometimes does not give you the best result.  What I do is that I evaluate all checkpoints from 60000 steps, and evaluate at least three times for each checkpoint.

**Comment by mlzxy** [comment]
BTW, what batch size do you use? I trained this model with 4x 40GB A100s. Using 2x 4090s may mean a smaller batch size?

**Comment by mlzxy** [comment]
By default, I use 4 GPUs and each GPU has a batch size of 48. The learning rate is the config is per sample as shown in [here](https://github.com/mlzxy/arp/blob/1de00688c4f126849f2b66f5f5db0bc7e104f822/rlb/train.py#L81).  Therefore, if you use a smaller batch size, you need to increase the learning rate accordingly.

**Comment by pussinmood** [comment]
Thanks a lot for your advice. I will try this out. Have a good day

**Comment by Holmes-GU** [comment]
> By default, I use 4 GPUs and each GPU has a batch size of 48. The learning rate is the config is per sample as shown in [here](https://github.com/mlzxy/arp/blob/1de00688c4f126849f2b66f5f5db0bc7e104f822/rlb/train.py#L81). Therefore, if you use a smaller batch size, you need to increase the learning rate accordingly. For example, if you use 2 GPUs with batch size of 24 on each, you need to multiply the current learning rate by 4.

Dear authors,

I use the default settings to train arp on aloha dataset. It seems to only use one GPU. However, the success rate on aloha is extremely low. I want to know where I got wrong and how could I improve the rate.

Really hope to get your advices.

<img width="1701" alt="image" src="https://github.com/user-attachments/assets/ebff8dd0-2d37-46a7-ac56-6d1c0771936f">
<img width="1709" alt="image" src="https://github.com/user-attachments/assets/54e41547-7f92-4b71-94e2-3cf1823624af">
<img width="1686" alt="image" src="https://github.com/user-attachments/assets/1d2c4645-3f0f-43fb-a456-16206a04ccdc">
<img width="1718" alt="image" src="https://github.com/user-attachments/assets/31e7f10e-954d-4ed3-a6a8-99c42325c17c">
<img width="1819" alt="image" src="https://github.com/user-attachments/assets/49f9eefc-57f9-4625-bbf9-afca8bb90895">
<img width="1731" alt="image" src="https://github.com/user-attachments/assets/dd9329b2-0631-45a2-9fdb-ce7e521fe253">
<img width="1695" alt="image" src="https://github.com/user-attachments/assets/bc45de02-38e7-4b1d-aa8f-5a68c0eadcd1">
<img width="1725" alt="image" src="https://github.com/user-attachments/assets/a896af10-0fe2-4491-abee-615d95b8591e">
<img width="1717" alt="image" src="https://github.com/user-attachments/assets/9dc67ef6-1535-4343-8758-c284fd71be78">
<img width="1754" alt="image" src="https://github.com/user-attachments/assets/a4b6d2c6-8bac-44b6-9a08-22b09fe4a15c">

**Comment by Walter0807** [comment]
> Thanks for your sharing spirit and remarkable work! I tried to train the arp+ model with two 4090, but when I evaluated it, the success rate just reached 79% total， and peg insertion was just 40%， just in the middle of the rvt2 and the arp. I want to know where I got wrong and how could I improve the rate. Really hope to get your advices.

I encountered the same situation.  

I ran the following command on two 80G GPUs:  

```bash
python3 train.py config=./configs/arp_plus.yaml hydra.job.name=arp_plus train.num_gpus=2 train.bs=96
```

The performance was still below expectations, especially on the peg insertion task.  

Additionally, I tried using six 32G GPUs with the following command:  

```bash
python3 train.py config=./configs/arp_plus.yaml hydra.job.name=arp_plus train.num_gpus=6 train.bs=32
```

The results were better but still slightly lower than the released checkpoint.  

Any suggestions or thoughts would be greatly appreciated!

**Comment by Walter0807** [comment]
> > Thanks for your sharing spirit and remarkable work! I tried to train the arp+ model with two 4090, but when I evaluated it, the success rate just reached 79% total， and peg insertion was just 40%， just in the middle of the rvt2 and the arp. I want to know where I got wrong and how could I improve the rate. Really hope to get your advices.
> 
> I encountered the same situation.
> 
> I ran the following command on two 80G GPUs:
> 
> python3 train.py config=./configs/arp_plus.yaml hydra.job.name=arp_plus train.num_gpus=2 train.bs=96
> The performance was still below expectations, especially on the peg insertion task.
> 
> Additionally, I tried using six 32G GPUs with the following command:
> 
> python3 train.py config=./configs/arp_plus.yaml hydra.job.name=arp_plus train.num_gpus=6 train.bs=32
> The results were better but still slightly lower than the released checkpoint.
> 
> Any suggestions or thoughts would be greatly appreciated!

I can reproduce the performance now, thanks!
